conjur/user/user_logic.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+
+"""
+UserLogic module
+
+This module is the business logic for handling all user-related activity
+"""
+
+# Builtins
+import logging
+
+# Internals
+from conjur.credentials_data import CredentialsData
+
+class UserLogic:
+    """
+    UserLogic
+
+    This class holds the business logic for handling
+    user activity
+    """
+    def __init__(self, conjurrc_data, credentials_from_store, client, resource):
+        self.conjurrc_data=conjurrc_data
+        self.credentials_from_store=credentials_from_store
+        self.client=client
+        self.resource=resource
+
+    # pylint: disable=logging-fstring-interpolation
+    def rotate_api_key(self, resource_to_update):
+        """
+        Method to trigger two types of API key actions.
+        1. The user can either rotate their own API key or
+        2. Rotate another users API key
+        """
+        loaded_credentials = self.extract_credentials_from_credential_store()
+        user_from_credential_store = loaded_credentials['login_id']
```

Comment:
login_id  is better to be strong typed and not extracted as a key from loaded_credentials.
if so i suggest put 'login_id' as a const somewhere and not as a magic string

Reply:
do you have an example of what you are suggesting? B/c the value of the const depends on the above line,
`loaded_credentials = self.extract_credentials_from_credential_store()`

Reply:
No example, just it's preferred to not use magic strings, and rather to const it before
https://softwareengineering.stackexchange.com/questions/365339/what-is-wrong-with-magic-strings

Reply:
Why don't you use an object with fields instead of map? Then you would have `credentials.login_id`